### PR TITLE
Hkpyo Retrofit

### PR DESCRIPTION
**Pull request by srfiorini**
Retrofited implementation of hkpyo to work on master version of hkpy.

### CODE REVIEW

**Comment by raphael-melo** [comment]
Just did a quick check of behaviour against a known to be working set of unit tests in a system that used the previous version of this library. Everything is behaving as expected. Now, I'll look through the changes to see if I can catch anything clearly odd or wrong.

File: requirements.txt [hunk 1]
```diff
@@ -1 +1,5 @@
-requests
\ No newline at end of file
+certifi==2019.11.28
+chardet==3.0.4
+idna==2.9
+requests==2.23.0
+urllib3==1.25.8
```
Comment (raphael-melo) [inline requirements.txt:5]: Is this version of urllib necessary? `urllib` is already one of Python 3's standard libraries.

File: hkpy/hklib/entity.py [hunk 2]
```diff
@@ -84,12 +84,35 @@ def to_dict(self) -> Dict:
...
+        super().__init__(type_, id_, properties=properties, metaproperties=metaproperties)
+        self.parent = parent
+
+    def to_dict(self) -> Dict:
+        jobj = super().to_dict()
+
+        if self.parent:
+            jobj['parent'] = self.parent
+        else:
+            jobj['parent'] = 'null'
```
Comment (vitornl) [inline hkpy/hklib/entity.py:116]: When null parent must be None, i.e., `jobj['parent'] = None`

File: hkpy/hklib/entity.py [hunk 3]
```diff
@@ -84,12 +84,35 @@ def to_dict(self) -> Dict:
                 jobj['type'] = v.value
             elif k == 'class_name':
                 jobj['className'] = v
+            elif k == 'properties':
+                jobj['properties'] = v
             elif k == 'metaproperties':
                 jobj['metaProperties'] = v
-            else:
```
Comment (vitornl) [inline hkpy/hklib/entity.py:89]: There is a huge problem in removing this else. All HKEntity attributes that are no specified in the above if-elif are now ignored. For example: `interfaces`, `binds`, among others.

**Pull request merged**